server-spi-private/src/main/java/org/keycloak/authorization/store/PermissionTicketStore.java

```diff
@@ -33,10 +33,20 @@
  */
 public interface PermissionTicketStore {
 
+    /**
+     * Legacy store doesn't store realm id for any entity and no method there is using new introduced RealmModel parameter.
+     * The parameter was introduced for usage only in the new storage. Therefore, in some cases we may break our rule specified in JavaDoc
+     * and use {@code null} value as parameter that otherwise cannot be {@code null}. We need to be careful and place such value only to a method call
+     * that cannot end up in the new store because it would end with {@link NullPointerException}. To mark all places where we do this,
+     * we use this variable so it is easily searchable.
+     */
+    static RealmModel NULL_REALM = null;
```

Comment:
You are right. This should not be here, it is quite confusing. Maybe we should not have it in the `server-spi*` modules at all. A good place would be the module that will be introduced here: https://github.com/keycloak/keycloak/issues/10279, however this is not there yet. I think it would make sense to have it inside `model/[jpa|infinispan]` to make sure we do not use it anywhere else. WDYT?

Reply:
It makes sense to me, I'd add a note to https://github.com/keycloak/keycloak/issues/10279 (as part of it) to move it into the new module. For now I believe it's ok to have it in `infinispan` module.

Reply:
Thinking about this again. Maybe it makes even more sense to have it in `infinispan/jpa` modules rather than in the legacy. Let's leave it as is. 

Please let me know once you see and agree with the commit I added so I can squash them.

Reply:
> Please let me know once you see and agree with the commit I added so I can squash them.

Please, go ahead, The PR LGTM!